Functions/Mock.ps1

```diff
@@ -1635,11 +1635,23 @@ function Reset-ConflictingParameters {
 }
 
 $script:ConflictingParameterNames = @(
-    "PSEdition"
+    #
 )
 
 function Get-ConflictingParameterNames {
```

Comment:
Let's change the function goal a little bit here. Instead of just finding at runtime the read-only/constant variables, assign them to the original `$script:ConflictingParameterNames` variable.

With this change, we should change the function verb, should we use `Set-ConflictingParameterNames` or `Resolve-ConflictingParameterNames`?

Reply:
It is still a `Get` you are getting the data and then saving it somewhere. `Set` would mean to set the variables in the system, and `Resolve` would mean that we are resolving it from something. I think `Get` is okay. :)

Reply:
So you're thinking to to move `Get-ConflictingParamterNames` somewhere where it will get called when `Mock` is called, and have it set the `$script:ConflictingParameterNames` variable, then replace the two current instances of `Get-ConflictingParamterNames` with a call to the variable directly?

In that case `Get-ConflictingParamterNames` wouldn't be returning anything, and would instead be setting the variable, so I would think `Set-ConflictingParamterNames` would be more appropriate, but maybe I'm misunderstanding the change.

Reply:
@renehernandez I see in another comment you mentioned calling this in Pester.psm1, I thought about that, but I'm not sure if that'd be entirely safe.  In theory Constant and ReadOnly variables are the automatic variables and they won't change, it is possible that a new automatic variable could be instantiated between loading the Pester module and running a test.  Additionally, the `New-Variable` command can create ReadOnly and Constant variables, so a user could indiscriminately be adding these types of potentially conflicting variables to their environment at anytime, so the safest time to look for them is when you call a Mock I would think

Reply:
I was thinking about it like this: `$script:ConflictingParameterNames = Get-ConflictingParamterNames` and then just refer to the script scoped variable anywhere else. So the function would still be a `get` function. No need to explicitly set the variable inside of the function. It is possible to do, but it is a very non-standard approach.

--- 

I don't think that detecting the read-only/constant variables at runtime is a good idea, it would be a very systematic solution, but the conflicts are rare, and we need this fix for variables that are built-in but not necessarily for the variables that user introduces. My main concern is performance of this solution, Mock is slow as it is, and searching through the variables in clean session takes some 3-8ms on my computer (ps 6 on macos), that is quite a lot to add to every mock declaration even though it is very unlikely to have this conflict.